components/raftstore/src/store/async_io/read.rs

```diff
@@ -141,11 +168,92 @@ where
                     },
                 );
             }
-            ReadTask::GenTabletSnapshot { region_id, .. } => {
-                // TODO: implement generate tablet snapshot for raftstore v2
-                self.notifier
-                    .notify_snapshot_generated(region_id, Box::new(RaftSnapshot::default()));
+
+            ReadTask::GenTabletSnapshot {
+                region_id,
+                tablet,
+                region_state,
+                last_applied_term,
+                last_applied_index,
+                canceled,
+                for_balance,
+            } => {
+                SNAP_COUNTER.generate.start.inc();
+                if canceled.load(Ordering::Relaxed) {
+                    info!("generate snap is canceled"; "region_id" => region_id);
+                    SNAP_COUNTER.generate.abort.inc();
+                    return;
+                }
+                let start = Instant::now();
+                let _io_type_guard = WithIoType::new(if for_balance {
+                    IoType::LoadBalance
+                } else {
+                    IoType::Replication
+                });
+                // the state should already checked in apply workers.
+                assert_ne!(region_state.get_state(), PeerState::Tombstone);
+                let key = SnapKey::new(region_id, last_applied_term, last_applied_index);
+                self.snap_mgr().register(key.clone(), SnapEntry::Generating);
+                defer!(self.snap_mgr().deregister(&key, &SnapEntry::Generating));
+                let mut snapshot = Snapshot::default();
+                // Set snapshot metadata.
+                snapshot.mut_metadata().set_index(key.idx);
+                snapshot.mut_metadata().set_term(key.term);
+                let conf_state = util::conf_state_from_region(region_state.get_region());
+                snapshot.mut_metadata().set_conf_state(conf_state);
+                // Set snapshot data.
+                let mut snap_data = RaftSnapshotData::default();
```

Comment:
should we increase the SNAPSHOT_VERSION since the snapshot data's format is changed.

Reply:
In this PR #13640, I have mark this version is SNAPSHOP_VERSION_V2(3) to decide how to send/recv snapshot directory?

Reply:
> In this PR #13640, I have mark this version is SNAPSHOP_VERSION_V2(3) to decide how to send/recv snapshot directory?

Got it.